flocker/node/agents/test/test_blockdevice.py

```diff
@@ -428,37 +428,44 @@ def test_non_manifest_dataset(self):
             expected_changes,
         )
 
-    def attached_dataset_test(self, state):
+    def attached_dataset_test(self, state, pass_device_path=True):
```

Comment:
I think with the new argument, test_non_manifest_dataset can use this helper as well and set it to False.

Probably should rename it to non_manifest_dataset_test and move it above the preceding test in that case.